spec/rubocop/magic_comment_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe RuboCop::MagicComment do
+  shared_examples 'magic comment' do |comment, expectations = {}|
+    encoding = expectations[:encoding]
+    frozen_string = expectations[:frozen_string_literal]
+
+    it "returns #{encoding.inspect} for encoding when comment is #{comment}" do
+      expect(described_class.parse(comment).encoding).to eql(encoding)
```

Comment:
I use `eql` instead of `eq` when possible. It is a more strict matcher since `eql?` indicates comparison without coercion